### ISSUE TITLE
Error with ./kustomize build overlays/local-production-local/ | $KUBECTL apply -f -

### ISSUE THREAD

**Issue by 211and300b** [wl-net/arcus-k8#2]
Hello,

When I am running setup-local.sh on an CentOS7 bare metal, I got stuck at this code block in setup-local.sh:

function apply {
  ./kustomize build overlays/local-production-local/ | $KUBECTL apply -f -
}
retry 10 apply

The errors are:
error: no objects passed to apply
Retry 1/10 exited 1, retrying in 1 seconds...
Error: AccumulateTarget: rawResources failed to read Resources: GroupVersionKindName: resid.ResId{gvKind:gvk.Gvk{Group:"networking.istio.io", Version:"v1alpha3", Kind:"DestinationRule"}, name:"cassandra", prefix:"", suffix:"", namespace:"default"} already exists b the map

Eventually, it would exhaust all retries and abort.  Is this something that I should try differently?

Thank you!

**Comment by AndrewX192** [comment]
Looks like a bug to me. I've been struggling to get istio to work with cassandra (a more general problem than anything specific to Arcus), and thus I think I ended up creating a duplicate destinationrule that caused the problem for you. I've removed this in the latest revision.

**Comment by 211and300b** [comment]
After apply the patch, a different error shows at the same kustomize line.   I could reproduce this on CentOS and uBuntu too.

May I ask what hosting OS did you use (photon, coreOS, etc)?   Any extra configuration required before running this setup script?

Thank you!


===  microk8s.kubectl get events ===
2m49s       Warning   InvalidDiskCapacity       node/ubuntu-k8   invalid capacity 0 on image filesystem

=== setup.local.sh errors below ===

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8e851]

goroutine 1 [running]:
sigs.k8s.io/kustomize/k8sdeps/transformer/patch.(*transformer).Transform(0xc0002c6dc0, 0x12a2ee0, 0xc0003ab000, 0x7f8488398008, 0x41379a)
        /go/src/sigs.k8s.io/kustomize/k8sdeps/transformer/patch/transformer.go:51 +0x341
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).transform(0xc0000db140, 0x12a2ee0, 0xc0003ab000, 0x1, 0x6)
        /go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:60 +0x79
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).Transform(0xc0000db140, 0x12a2ee0, 0xc0003ab000, 0x6, 0xc000597b60)
        /go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:56 +0x84
sigs.k8s.io/kustomize/pkg/accumulator.(*ResAccumulator).Transform(...)
        /go/src/sigs.k8s.io/kustomize/pkg/accumulator/resaccumulator.go:133
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).runTransformers(0xc0002dc600, 0xc0002b11d0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:322 +0x3fb
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).AccumulateTarget(0xc0002dc600, 0xc000074620, 0xc0000be210, 0xc0002e7c78)
        /go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:241 +0x2b6
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).makeCustomizedResMap(0xc0002dc600, 0x1, 0xc000482450, 0x1251e80, 0x1b20538, 0xc00036f930)
        /go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:126 +0x2f
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).MakeCustomizedResMap(...)
        /go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:117
sigs.k8s.io/kustomize/pkg/commands/build.(*Options).RunBuild(0xc0002b0060, 0x1251d40, 0xc00000e018, 0x127ecc0, 0x1b20538, 0x12947c0, 0x1b20538, 0xc000482450, 0x1251e80, 0x1b20538, ...)
        /go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:130 +0x145
sigs.k8s.io/kustomize/pkg/commands/build.NewCmdBuild.func1(0xc0000c9b80, 0xc00036fae0, 0x1, 0x1, 0x0, 0x0)
        /go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:80 +0x193
github.com/spf13/cobra.(*Command).execute(0xc0000c9b80, 0xc00036fa90, 0x1, 0x1, 0xc0000c9b80, 0xc00036fa90)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:756 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000c8f00, 0xec29d4, 0xc0002e7f88, 0xc000094058)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:846 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:794
main.main()
        /go/src/sigs.k8s.io/kustomize/cmd/kustomize/main.go:13 +0x28
error: no objects passed to apply
Retry 5/10 exited 1, retrying in 16 seconds...

**Comment by AndrewX192** [comment]
I used Ubuntu 18.04, and I actually recently installed Ubuntu 18.04 and tested it. I also experienced the "invalid capacity 0" error, but increasing disk space to around 25GB fixed that problem. I'm not entirely sure what your second problem is, but I will note that version of kustomize (and all of the docker containers, libraries, etc.) are built for amd64. If you are using a 32bit system or ARM, this project is not going to work as-is. I don't plan on adding support for 32bit systems, since that seems like a dead-end, and ARM support doesn't seem especially useful given that Arcus has very high memory requirements at this time, well beyond what a raspberry pi (or cluster of them) will be able to provide per-process.
